numpy/lib/function_base.py

```diff
@@ -115,13 +124,28 @@ def scott(x):
         Scott Estimator
         The binwidth is proportional to the standard deviation of the data and
         inversely proportional to the cube root of data size (asymptotically optimal)
-
         """
         h = 3.5 * x.std() * x.size ** (-1.0 / 3)
         if h > 0:
             return np.ceil(x.ptp() / h)
         return 1
 
+    def doane(x):
+        """
+        Doane's Estimator
+        Improved version of Sturges' formula which works better for non-normal
+        data.
+        See http://stats.stackexchange.com/questions/55134/doanes-formula-for-histogram-binning
+        """
+        if x.size > 2:
+            stdskew = np.sqrt(6.0 * (x.size - 2) / ((x.size + 1.0) * (x.size + 3)))
+            sigma = np.std(x)
+            if sigma > 0:
+                mu = np.mean(x)
+                skew = (np.mean(x**3) - 3 * mu * sigma**2 - mu**3) / sigma**3
```

Comment:
I got the Doane '76 paper, and can look at it tomorrow.
However, the one of the first links searching for it in Google found this http://www.robjhyndman.com/papers/sturges.pdf
So it might only be "Aesthetic" in small sample sizes that were usual in 1976.